modules/gateways/blockonomics/payment.php

```diff
@@ -24,54 +24,47 @@
 /*
  * SET POST PARAMETERS TO VARIABLES AND CHECK IF THEY EXIST
  */
-$get_order = htmlspecialchars(isset($_REQUEST['get_order']) ? $_REQUEST['get_order'] : '');
-$finish_order = htmlspecialchars(isset($_REQUEST['finish_order']) ? $_REQUEST['finish_order'] : '');
-
-$order_hash = htmlspecialchars(isset($_REQUEST['order']) ? $_REQUEST['order'] : '');
-
-$system_url = App::getSystemURL();
+$show_order = isset($_GET["show_order"]) ? htmlspecialchars($_GET['show_order']) : "";
+$crypto = isset($_GET["crypto"]) ? htmlspecialchars($_GET['crypto']) : "";
+$select_crypto = isset($_GET["select_crypto"]) ? htmlspecialchars($_GET['select_crypto']) : "";
+$finish_order = isset($_GET["finish_order"]) ? htmlspecialchars($_GET['finish_order']) : "";
+$get_order = isset($_GET['get_order']) ? htmlspecialchars($_GET['get_order']) : "";
+
+if($crypto === "empty"){
+    $ca->setTemplate('/modules/gateways/blockonomics/assets/templates/no_crypto_selected.tpl');
+}else if ($show_order && $crypto) {
+    $time_period_from_db = $blockonomics->getTimePeriod();
+    $time_period = isset($time_period_from_db) ? $time_period_from_db : '10';
+    $ca->assign('time_period', $time_period);
+    $active_currencies = $blockonomics->getActiveCurrencies();
+    $ca->assign('active_currencies', json_encode($active_currencies));
+    $ca->assign('selected_crypto', $crypto);
+    $ca->assign('order_uuid', $show_order);
+    
+    $ca->setTemplate('/modules/gateways/blockonomics/assets/templates/checkout.tpl');
+}else if ($select_crypto) {
+    $active_currencies = $blockonomics->getActiveCurrencies();
+    $ca->assign('active_currencies', $active_currencies);
+    $ca->assign('order_hash', $select_crypto);
+    $ca->setTemplate('/modules/gateways/blockonomics/assets/templates/crypto_options.tpl');
+}else if ($finish_order) {
+    $finish_url = App::getSystemURL() . 'viewinvoice.php?id=' . $finish_order . '&paymentsuccess=true';
+    header("Location: $finish_url");
+    exit();
+}else if ($get_order && $crypto) {
+    $existing_order = $blockonomics->processOrderHash($get_order, $crypto);
```

Comment:
Lets refactor this to match woocommerce by moving functionality into the Blockonomics class:
```
if($crypto === "empty"){
    $blockonomics->load_blockonomics_template('no_crypto_selected');
}else if ($show_order && $crypto) {
    $blockonomics->load_checkout_template($show_order, $crypto);
}else if ($select_crypto) {
    $blockonomics->load_blockonomics_template('crypto_options');
}else if ($finish_order) {
    $blockonomics->redirect_finish_order($finish_order);
}else if ($get_order && $crypto) {
    $existing_order = $blockonomics->processOrderHash($get_order, $crypto);
```